e2e-tests/cypress/support/page-objects/management/Reports.ts

```diff
@@ -0,0 +1,118 @@
+import { KibanaNavigation } from '../KibanaNavigation';
+import semver from "semver/preload";
+import { getKibanaVersion } from "../../helpers";
+
+export class Reports {
+  static navigateTo() {
+    cy.log('navigateTo');
+
+    // Check if we are already on the Stack Management page
+    cy.url().then(url => {
+      if (!url.includes('/app/management')) {
+        // If not, open Kibana navigation and click on Stack Management
+        KibanaNavigation.openKibanaNavigation();
+        cy.contains('Stack Management').click();
+      }
+
+      // Now, click on Reporting (whether we were already on Stack Management or not)
+      cy.contains('Reporting').click();
+    });
+  }
+
+  static waitForAllReportsToBeDone(reportTitle: string = 'Untitled discover search') {
+    cy.log('waitForAllReportsToBeDone');
+
+    cy.get('tr[data-test-subj="reportJobRow"]')
+      .each($reportItem => {
+        cy.wrap($reportItem)
+          .find('[data-test-subj="reportingListItemObjectTitle"]')
+          .should('contain.text', reportTitle);
+
+        cy.wrap($reportItem)
+          .find('[data-test-subj="reportJobStatus"]')
+          .contains(/Done|Completed/, {timeout: 60000})
+          .should('be.visible');
+
+        // Click the download button
+        if (semver.lte(getKibanaVersion(), '8.8.0')) {
+          cy.wrap($reportItem)
+            .find('[data-test-subj="reportJobActions"] button')
+            .first()
+            .click();
+        } else {
+          cy.wrap($reportItem)
+            .find('[data-test-subj="reportDownloadLink"]')
+            .first()
+            .click();
+        }
+
+        // Assert that a file is downloaded
+        cy.verifyDownload(`${ reportTitle }.csv`, {timeout: 10000}) // Adjust timeout if needed
+          .then(filePath => {
+            // Read the downloaded file content and assert it's not empty
+            cy.readFile(filePath)
+              .should('not.be.empty')
+              .then(fileContent => {
+                const rows = fileContent.split('\n');
+                expect(rows.length).to.be.greaterThan(1);
+              });
+          });
+      })
+      .then($matchingReports => {
+        expect($matchingReports).to.have.length(10);
+      });
+  }
+
+
+  static verifyNoReports() {
+    cy.log('verifyNoReports');
+
+    // Assert that there are no report rows present
+    cy.get('tr[data-test-subj="reportJobRow"]').should('not.exist');
+  }
+
+  static deleteAllReports(reportTitle: string = 'Untitled discover search') {

```

Comment:
This logic seems super complicated and fragile based on the Kibana version. Maybe we can just use the curl command to remove the reporting index? 
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/876c4b88-8885-433b-9994-40b8cc261502">

Reply:
@lodyczekoladowe What about this comment?

Reply:
> It seems to me that it is not possible to delete these reports using the curl command.

It should be possible (but also complicated). My proposal, was, to delete the reporting index by the Kibana Api
https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-index.html, thanks to it, all existing reports will be cleared

Reply:
That command delate index but don't delate reports. There is no curl command that deletes reports.

Reply:
When you remove an .`.reporting` index that stores reports, you also remove reports :)